lib/elixir/lib/module.ex

```diff
@@ -1158,6 +1273,18 @@ defmodule Module do
         :ok
     end
 
+    case {key, value} do
+      {:impl, nil} -> :ok
+      {:impl, false} -> :ok
+      {:impl, true} -> raise "TODO: implement"
```

Comment:
@josevalim The original proposal specified an `@impl true` variant but I actually prefer the explicitness of `@impl Module`.

Perhaps we could ship the specific case and see how people like it. If it seems obvious that `@impl true` would be nice to have, we can easily add it later. Adding it now and removing it later would be much more difficult.